src/main/java/org/jabref/migrations/PreferencesMigrations.java

```diff
@@ -407,6 +408,29 @@ static void upgradeColumnPreferences(JabRefPreferences preferences) {
         }
     }
 
+    /**
+     * Adds the search rank column to the main table if it is not already present.
+     * This column is necessary for sorting the table in floating mode.
+     */
+    static void addSearchRankColumn(JabRefPreferences preferences) {
+        List<String> columnNames = new ArrayList<>(preferences.getStringList(JabRefPreferences.COLUMN_NAMES));
+        if (!columnNames.contains(MainTableColumnModel.Type.SEARCH_RANK.getName())) {
+            List<String> sortOrder = new ArrayList<>(preferences.getStringList(JabRefPreferences.COLUMN_SORT_ORDER));
+            List<String> columnSortTypes = new ArrayList<>(preferences.getStringList(JabRefPreferences.COLUMN_SORT_TYPES));
+            List<String> columnWidths = new ArrayList<>(preferences.getStringList(JabRefPreferences.COLUMN_WIDTHS));
+
+            columnNames.addFirst(MainTableColumnModel.Type.SEARCH_RANK.getName());
+            sortOrder.addFirst(MainTableColumnModel.Type.SEARCH_RANK.getName());
+            columnSortTypes.addFirst(TableColumn.SortType.ASCENDING.toString());
+            columnWidths.addFirst(String.valueOf(ColumnPreferences.DEFAULT_COLUMN_WIDTH));
+
+            preferences.putStringList(JabRefPreferences.COLUMN_NAMES, columnNames);
+            preferences.putStringList(JabRefPreferences.COLUMN_SORT_ORDER, sortOrder);
+            preferences.putStringList(JabRefPreferences.COLUMN_SORT_TYPES, columnSortTypes);
+            preferences.putStringList(JabRefPreferences.COLUMN_WIDTHS, columnWidths);
+        }
+    }
+
```

Comment:
Please add some information about the Version. We also use the high level methods to create the new preferences, to keep this forward compatible with later versions.
Also: Should the search rank column always (!) be added to the List of columns? I believe only when the user uses floating mode to search. But floating mode is disabled by default. So I think this migration is probably unnecessary.

Reply:
Fixed in: [`4608d0c` (#11510)](https://github.com/JabRef/jabref/pull/11510/commits/4608d0cdd32f2a299a0b10a29562733a966b4e79)